envoy/stats/stats.h

```diff
@@ -218,5 +218,44 @@ using SizeFn = std::function<void(std::size_t)>;
  */
 template <typename Stat> using StatFn = std::function<void(Stat&)>;
 
+/**
+ * Interface for stats lazy initialization.
+ * To save memory and CPU consumption from unused stats, Envoy can enable the bootstrap config
```

Comment:
suggest slightly more detail:

To save memory and CPU consumption on blocks of stats that are never referenced throughout the process lifetime, they can be encapsulated in a DeferredCreationCompatibleInterface. Then the Envoy bootstrap configuration can be set to defer the instantiation of those block.  Note that when the blocks of stats are created, they carry an extra ~160 byte overhead (depending on worker thread count) due to  internal bookkeeping data structures. The overhead when deferred stats are disabled is just 8 bytes.

WDYT? You should confirm the 160 bytes.

Reply:
updated. 

did you see this?  https://github.com/envoyproxy/envoy/pull/23921#issuecomment-1584575166 

"The last column shows the overhead is around 5MB for 100K clusters."